### PR TITLE
refactor: Move code from UsersResponder to UserService and UserRestService

### PR DESCRIPTION
**Pull request by seakayone**
- **Move add groups methods from UsersResponder to RestService**
- **Move validation business logic from RestService to UserService**
- **Merge UserResponseADM with UserOperationResponseADM**
- **fmt**

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

**Review by mpro7 — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala [hunk 1]
```diff
@@ -84,15 +88,49 @@ case class UserService(
...
+      isInProject = update.projects.getOrElse(kUser.isInProject).distinct,
+      isInProjectAdminGroup = update.projectsAdmin.getOrElse(kUser.isInProjectAdminGroup).distinct,
+      isInGroup = update.groups.getOrElse(kUser.isInGroup).distinct,
       isInSystemAdminGroup = update.systemAdmin.getOrElse(kUser.isInSystemAdminGroup)
     )
     userRepo.save(updatedUser)
   }
 
-  private def toUser(kUser: KnoraUser): Task[User] = for {
+  def addGroupToUserIsInGroup(user: KnoraUser, group: GroupADM): IO[UserServiceError, KnoraUser] = for {
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala:99]: **suggestion (non-blocking):** Rename these functions.  
I know, this has been like that previously. But I find the naming of these functions unclear and confusing at best.

I would propose something along the lines of:

- `addGroupToUserIsInGroup` -> `addUserToGroup`
- `addProjectToUserIsInProject` -> `addUserToProject` or `makeProjectMember`
- `addProjectToUserIsInProjectAdminGroup` -> `addUserToProjectAdmin` or `makeUserProjectAdmin`

**Review by BalduinLandolt — APPROVED** [review]
LGTM, added one suggestion (non-blocking)

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala (hunk 1 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala:99]: The naming of these methods are actually clearer than your proposals. It reads a bit clunky but follows the model and naming.

Let me explain:
1. `Group`'s do not have members attached to them, the relationship is the other way round.
2. A user has an `IsInGroup` property which contains all `GroupIri`s the `User` belongs to.

So the current name does exactly what it says: 
* `addGroupToUserIsInGroup`
* add a `Group` to the `User`'s `IsInGroup` property.

Comment (BalduinLandolt) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala:99]: That was clear to me, but I don't like it. A function name IMO should be clear and easy to understand, and tell me what the function does on a high level, and not be concerned with the implementation details of how it is modelled under the hood.

But notice the `non-blocking` - if you want to keep it as is, that's ok.

**Review by siers — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala (hunk 1 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala:99]: You have convinced me:  a cleaner interface for the `UserService` is actually welcome. I will tackle this in my next PR.

Comment (BalduinLandolt) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/UserService.scala:99]: cool, looking forward to it :)

**Pull request merged**